Sources/NIOCore/Docs.docc/swift-concurrency.md

```diff
@@ -0,0 +1,244 @@
+# NIO and Swift Concurrency
+
+This article explains how to interface between NIO and Swift Concurrency 
+
+NIO was created before native Concurrency support in Swift existed, hence, NIO had to solve
+a few problems that have solutions in the language today. Since, the introduction of Swift Concurrency
+NIO has added numerous features to make the inter-op between NIOs channel eventing system and Swift's
+Concurrency primitives as easy as possible.
```

Comment:
```suggestion
This article explains how to interface between NIO and Swift Concurrency.

NIO was created before native Concurrency support in Swift existed, hence, NIO had to solve
a few problems that have solutions in the language today. Since the introduction of Swift Concurrency,
NIO has added numerous features to make the inter-op between NIO's channel event system and Swift's
Concurrency primitives as easy as possible.
```